persistent/persistence.py

```diff
@@ -11,22 +11,30 @@
 # FOR A PARTICULAR PURPOSE.
 #
 ##############################################################################
+import functools
 import struct
 
 from zope.interface import implementer
-
-from persistent.interfaces import IPersistent
-from persistent.interfaces import GHOST
-from persistent.interfaces import UPTODATE
-from persistent.interfaces import CHANGED
-from persistent.interfaces import STICKY
-
+from persistent import interfaces
 from persistent.interfaces import SERIAL_TYPE
 from persistent.timestamp import TimeStamp
 from persistent.timestamp import _ZERO
 from persistent._compat import copy_reg
 from persistent._compat import intern
+from persistent._compat import use_c_impl
+
+# We use implementation details of PickleCachePy
+# pylint:disable=protected-access
+# we have lots of not-quite-correct continuation indentation.
+# TODO: Fix that in a whitespace-only commit.
+# pylint:disable=bad-continuation
+# There are a few places we need to work with exact types.
+# pylint:disable=unidiomatic-typecheck
 
+GHOST = use_c_impl(interfaces.GHOST, 'GHOST')
+UPTODATE = use_c_impl(interfaces.UPTODATE, 'UPTODATE')
+CHANGED = use_c_impl(interfaces.CHANGED, 'CHANGED')
+STICKY = use_c_impl(interfaces.STICKY, 'STICKY')
```

Comment:
This looks like magic that could use an explanatory comment?

Reply:
Also, the Python versions are already imports from the C versions, when those are available?